lib/plugins/aws/package/compile/events/iotFleetProvisioning/index.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+const _ = require('lodash');
+
+class AwsCompileIotFleetProvisioningEvents {
+  constructor(serverless) {
+    this.serverless = serverless;
+    this.provider = this.serverless.getProvider('aws');
+
+    this.hooks = {
+      'package:compileEvents': this.compileIotFleetProvisioningEvents.bind(this),
+    };
+
+    this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'iotFleetProvisioning', {
+      type: 'object',
+      properties: {
+        enabled: { type: 'boolean' },
+        provisioningRoleArn: { $ref: '#/definitions/awsArn' },
+        templateBody: { type: 'object' },
+        templateName: { type: 'string' },
+      },
+      required: ['templateBody', 'provisioningRoleArn'],
+      additionalProperties: false,
+    });
+  }
+
+  compileIotFleetProvisioningEvents() {
+    this.serverless.service.getAllFunctions().forEach(functionName => {
+      const functionObj = this.serverless.service.getFunction(functionName);
+      let functionTemplateNumber = 0;
+
+      if (functionObj.events) {
+        functionObj.events.forEach(event => {
+          if (event.iotFleetProvisioning) {
+            functionTemplateNumber++;
```

Comment:
This approach is not safe for future changes. We already have reported cases with other events which are handled like that, that e.g. removing one event can break the deployment (indexes no longer match events they matched before).

I think that we should settle on some deterministic naming. Two ideas that come to my mind:
- Allow only one event per function 
- Require event `name` if more than one event is configured for single functions

Or maybe you have some better idea?

Reply:
Understood, let's lock it down to one event per function, I don't see any usecase where both provisioning template would use the exact same lambda.